ctapipe/tools/train_energy_regressor.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+
+from ctapipe.core import Tool
+from ctapipe.core.traits import Bool, Int, Path, flag
+from ctapipe.io import TableLoader
+from ctapipe.reco import CrossValidator, EnergyRegressor
+from ctapipe.reco.preprocessing import check_valid_rows
+
+__all__ = [
+    "TrainEnergyRegressor",
+]
+
+
+class TrainEnergyRegressor(Tool):
+    """
+    Tool to train a `~ctapipe.reco.EnergyRegressor` on dl1b/dl2 data.
+
+    The tool first performs a cross validation to give an initial estimate
+    on the quality of the estimation and then finally trains one model
+    per telescope type on the full dataset.
+    """
+
+    name = "ctapipe-train-energy-regressor"
+    description = __doc__
+
+    examples = """
+    ctapipe-train-energy-regressor \\
+        --config ml_config.yaml \\
+        --input gamma.dl2.h5 \\
+        --ouput energy_regressor.pkl
+    """
+
+    output_path = Path(
+        default_value=None,
+        allow_none=False,
+        directory_ok=False,
+        help="Ouput path for the trained reconstructor.",
+    ).tag(config=True)
+
+    n_events = Int(
+        default_value=None,
+        allow_none=True,
+        help=(
+            "Number of events for training the model."
+            " If not give, all available events will be used."
+        ),
+    ).tag(config=True)
+
+    overwrite = Bool(help="overwrite existing output files").tag(config=True)
+
+    random_seed = Int(
+        default_value=0, help="Random seed for sampling and cross validation"
+    ).tag(config=True)
```

Comment:
It's not clear here if the random seed value of 0 is special or not, or if there is an option to auto-generate one.   
Shouldn't the default be to auto-generate?

Reply:
Also, if auto-generated, it should somehow get written to the output (I guess it will be in the pickled class at least)

Reply:
There is no option to auto-generate one and I think the default should be to be reproducible. There is no reason for auto-generating one I think, since this is not a simulation, it's about splitting the events / sub-sampling them.

Reply:
0 is not special here